topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementRequestsHandler.java

```diff
@@ -83,4 +115,74 @@ private ActorFuture<TopologyChangeStatus> applyOperations(
 
     return responseFuture;
   }
+
+  private List<TopologyChangeOperation> generatePartitionDistributionOperations(
+      final ClusterTopology currentTopology, final Set<MemberId> brokers) {
+    final List<TopologyChangeOperation> operations = new ArrayList<>();
+
+    final var oldDistribution = TopologyUtil.getPartitionDistributionFrom(currentTopology, "temp");
+    // We assume that all partitions have the same replication factor
+    final int replicationFactor =
```

Comment:
:thought_balloon: Shouldn't we at least verify that the `replicationFactor` is achievable with the given number of brokers? I don't think `RoundRobinPartitionDistributor` handles this gracefully.

Reply:
:+1: We should do it.

Reply:
Done.